RIOT-OS/RIOT
# nrf52: Implement EasyDMA-based SPI peripheral implemenation

### Contribution description

This PR splits the common nrf5x SPI peripheral into the old (,and for the nrf52 deprecated peripheral,) nRF51 driver and a new EasyDMA-based driver for the nRF52. The new driver is significantly faster than the old one as the bytes can be send back-to-back without overhead from the CPU.

I decided to split the nrf5x common peripheral driver, it became too much of an ifdef hell to keep them merged.

The peripheral driver requires a GPIO interrupt on the nrf52832 due to [errata 58](https://infocenter.nordicsemi.com/index.jsp?topic=%2Ferrata_nRF52832_Rev2%2FERR%2FnRF52832%2FRev2%2Flatest%2Fanomaly_832_58.html&anchor=anomaly_832_58). The proposed driver uses an adapted form of the workaround presented by Nordic. This anomaly also exists on engineering sample A of the nrf52840. I decided not to support that to shave off some additional flash usage and runtime checks. This might be an issue on the nrf52840-pdk boards.

### Testing procedure

Compilation should verify that all nrf52 boards have been adapted, the board config requires modification from `NRF_SPI#` to `NRF_SPIM#`. Maybe @MrKevinWeiss can give this a try on the RobotFW tests to verify the runtime behaviour.

### Issues/PRs references

None
